tests/Avalonia.IntegrationTests.Appium/WindowTests_MacOS.cs

```diff
@@ -259,18 +255,19 @@ private IDisposable OpenWindow(PixelSize? size, ShowWindowMode mode, WindowStart
             return showButton.OpenWindowWithClick();
         }
 
-        private static int GetWindowOrder(IReadOnlyCollection<AppiumWebElement> elements, string identifier)
+        private AppiumWebElement GetWindow(string identifier)
         {
-            return elements.TakeWhile(x =>
-                x.FindElementByXPath("XCUIElementTypeWindow")?.GetAttribute("identifier") != identifier).Count();
+            // The Avalonia a11y tree currently exposes two nested Window elements, this is a bug and should be fixed 
+            // but in the meantime use the `parent::' selector to return the parent "real" window. 
+            return _session.FindElementByXPath(
```

Comment:
Also changed this to send an XPath query rather than doing multiple roundtrips to appium to find the window.